File: runtime/tools/world.py
"""Perception tools: get_world_state()."""
from __future__ import annotations

from typing import Any, Dict

from runtime.tools.registry import ToolContext, ToolRegistry, ToolSpec


GET_WORLD_STATE_SCHEMA: Dict[str, Any] = {
    "type": "object",
    "properties": {},
    "additionalProperties": False,
}


def get_world_state(context: ToolContext) -> Dict[str, Any]:
    return context.get_world_state()


def register_world_tools(registry: ToolRegistry) -> None:
    registry.register(
        ToolSpec(
            name="get_world_state",
            description="Return the NPC's current relevant world state.",
            parameters=GET_WORLD_STATE_SCHEMA,
            handler=get_world_state,
            category="perception",
        )
    )

File: runtime-go/internal/tools/world.go
package tools

import "github.com/ThanabordeeN/AgentRD2/runtime-go/internal/domain"

// getWorldStateSchema mirrors “GET_WORLD_STATE_SCHEMA“.
var getWorldStateSchema = map[string]any{
	"type":                 "object",
	"properties":           map[string]any{},
	"additionalProperties": false,
}

// getWorldState mirrors “runtime/tools/world.py::get_world_state“. The
// Python handler returns the world dict itself; “_call_tool“ then wraps it as
// “ToolResult.completed(request, result=<dict>)“, which is what
// completedData reproduces.
func getWorldState(ctx *Context, args map[string]any) domain.ToolResult {
	if worldOf(ctx) == nil {
		return failedCall("get_world_state", ctx, args, "world reader is not configured")
	}
	return completedData("get_world_state", ctx, args, ctx.GetWorldState())
}

// RegisterWorldTools registers the perception tool “get_world_state“.
func RegisterWorldTools(registry *Registry) {
	_ = registry.Register(Spec{
		Name:        "get_world_state",
		Description: "Return the NPC's current relevant world state.",
		Parameters:  getWorldStateSchema,
		Category:    "perception",
		Handler:     getWorldState,
	})
}
